app/models/field.rb

```diff
@@ -0,0 +1,19 @@
+class Field < ApplicationRecord
+  belongs_to :calculator
+  before_save :set_selector
+  validates :uuid, :type, :selector, :label, :kind, presence: true
+  enum kind: { form: 0, parameter: 1, result: 2 }
+
+  private
+    def set_selector
+      if self.selector.nil?
+        selected_rows_count = Field.where(type: :kind).count
+        if selected_rows_count.positive?
+          previous_number = Field.where(type: :kind).last[1]
+          self.selector = self.type[0].upcase + previous_number.next.to_s
+        else
+          self.selector = self.type[0].upcase + "1"
```

Comment:
`Field.where(type: :kind)` will always return an empty result because the query is incorrect. You need records with the same `kind` value.

Reply:
Thank you. I tried to change it to `self.kind`. Unfortunately I've not found in rails documentation how to query selected by user enum. I've found a lot of examples how query concrete enum option. But I don't know which is selected by user.